typescript/infra/scripts/deploy.ts

```diff
@@ -49,32 +49,25 @@ async function main() {
     multiProvider.setSigner(fork, signer);
   }
 
+  let config: any;
```

Comment:
ChainMap<any>?

How do we enforce that the config and the deployer types match?

Reply:
We could 
```
getConfigAndDeployer<C>(): { config: C, deployer extends HyperlaneDeployer<C, any>}
```

Reply:
we are doing that in `deployWithArtifacts`